src/daos_vol.c

```diff
@@ -1865,7 +1893,13 @@ H5_daos_get_conn_cls(void *item, H5VL_get_conn_lvl_t H5VL_DAOS_UNUSED lvl, const
  *---------------------------------------------------------------------------
  */
 static herr_t
-H5_daos_get_cap_flags(const void H5VL_DAOS_UNUSED *info, unsigned *cap_flags)
+H5_daos_get_cap_flags(const void H5VL_DAOS_UNUSED *info,
+#if H5VL_VERSION > 2
```

Comment:
Here too. I'll stop listing them. I think the check for H5VL_VERSION > 3 is fine since it's illustrating that 3 is the maximum.